### PR TITLE
Upgrade to bazel-2.0.0

### PR DESCRIPTION
**Pull request by elliottt**
### Motivation

We're on a pretty old version of bazel, this bumps us up to the current release.

### Test plan

CI

### CODE REVIEW

File: test/cli/cache-keeps-print-options/cache-keeps-print-options.out [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class ::<root> < ::Object () @ (... removed core rbi locs ..., https://github.com/sorbet/sorbet/tree/master/bazel-out/host/genfiles/rbi/procs.rbi#L1, test/cli/cache-keeps-print-options/cache-keeps-print-options.rb:1)
+class ::<root> < ::Object () @ (... removed core rbi locs ..., https://github.com/sorbet/sorbet/tree/master/bazel-out/host/bin/rbi/procs.rbi#L1, test/cli/cache-keeps-print-options/cache-keeps-print-options.rb:1)
```
Comment (jez) [inline test/cli/cache-keeps-print-options/cache-keeps-print-options.out]: Non-blocking suggestion: change the code that handles `censorForSnapshotTests` to hide / shorten the `procs.rbi` RBI file.

File: third_party/externals.bzl [hunk 2]
```diff
@@ -35,17 +35,20 @@ def register_sorbet_dependencies():
...
         name = "com_google_protobuf",
-        sha256 = "d7a221b3d4fb4f05b7473795ccea9e05dab3b8721f6286a95fffbffc2d926f8b",
-        strip_prefix = "protobuf-3.6.1",
-        urls = ["https://github.com/google/protobuf/archive/v3.6.1.zip"],
+        sha256 = "e4f8bedb19a93d0dccc359a126f51158282e0b24d92e0cad9c76a9699698268d",
+        strip_prefix = "protobuf-3.11.2",
+        urls = ["https://github.com/protocolbuffers/protobuf/archive/v3.11.2.zip"],
+        # sha256 = "d7a221b3d4fb4f05b7473795ccea9e05dab3b8721f6286a95fffbffc2d926f8b",
+        # strip_prefix = "protobuf-3.6.1",
+        # urls = ["https://github.com/google/protobuf/archive/v3.6.1.zip"],
```
Comment (jez) [inline third_party/externals.bzl]: What's the commented out code here for? Can we delete it?

File: third_party/externals.bzl [hunk 3]
```diff
@@ -123,43 +126,44 @@ def register_sorbet_dependencies():
     git_repository(
         name = "com_google_absl",
         remote = "https://github.com/abseil/abseil-cpp.git",
+        # commit = "63ee2f8877915a3565c29707dba8fe4d7822596a",
```
Comment (jez) [inline third_party/externals.bzl]: ditto

File: tools/toolchain/webasm-darwin/cc_toolchain_config.bzl [hunk 4]
```diff
@@ -0,0 +1,182 @@
+load(
```
Comment (jez) [inline tools/toolchain/webasm-darwin/cc_toolchain_config.bzl]: ooc, what are these two new large wasm build files for? did you copy them from somewhere?

File: third_party/ruby/ruby-2.6.BUILD [hunk 5]
```diff
@@ -19,13 +19,14 @@ config_setting(
     values = {"host_cpu": "k8"},
 )
 
-
```
Comment (jez) [inline third_party/ruby/ruby-2.6.BUILD]: Are the changes in these two ruby BUILD files mostly just buildifier formatting changes, or did you have to do a lot of work to get ruby to build under new bazel specifically?

File: tools/toolchain/webasm-darwin/cc_toolchain_config.bzl (hunk 4 above)
Comment (elliottt) [reply tools/toolchain/webasm-darwin/cc_toolchain_config.bzl]: They're the new way to describe a C++ toolchain to bazel. This is also why the CROSSTOOL files are removed.

File: third_party/ruby/ruby-2.6.BUILD (hunk 5 above)
Comment (elliottt) [reply third_party/ruby/ruby-2.6.BUILD]: There was only one significant change to each file, adding the `toolchains`  attribute to each  `run_configure` genrule:

https://github.com/sorbet/sorbet/pull/2444/files#diff-30490048ed9b3e2a33af671e6ad05b20R72-R75

File: third_party/externals.bzl (hunk 3 above)
Comment (elliottt) [reply third_party/externals.bzl]: Yes, sorry for leaving these in there, I'll remove them.

File: test/cli/cache-keeps-print-options/cache-keeps-print-options.out (hunk 1 above)
Comment (elliottt) [reply test/cli/cache-keeps-print-options/cache-keeps-print-options.out]: This is a great idea, I'll make a follow-up PR to address it.

File: tools/toolchain/webasm-darwin/cc_toolchain_config.bzl (hunk 4 above)
Comment (elliottt) [reply tools/toolchain/webasm-darwin/cc_toolchain_config.bzl]: I should also point out that I generated them by using a tool from rules_cc that converts from the old CROSSTOOL format.

**Pull request merged**